libafl_derive/src/lib.rs

```diff
@@ -69,3 +69,109 @@ pub fn libafl_serdeany_derive(input: TokenStream) -> TokenStream {
         libafl_bolts::impl_serdeany!(#name);
     })
 }
+
+/// Display macro to implement `Display` for a struct where all fields implement `Display`.
+/// The result is the concatenation of all fields display.
+/// Specifically handled cases:
+/// Options: Some => inner type display None => "".
+/// Vec: inner type display concatenated with spaces.
+/// Generics or other more or less exotic stuff are not supported.
+#[proc_macro_derive(Display)]
+pub fn libafl_display(input: TokenStream) -> TokenStream {
+    let DeriveInput { ident, data, .. } = parse_macro_input!(input as DeriveInput);
+
+    if let Struct(s) = data {
+        if let Named(fields) = s.fields {
+            let vec_fields = fields
+                .named
+                .iter()
+                .filter(|it| libafl_display_type(&it.ty) == TyVec)
+                .map(|it| &it.ident);
+            let options_fields = fields
+                .named
+                .iter()
+                .filter(|it| libafl_display_type(&it.ty) == TyOption)
+                .map(|it| &it.ident);
+            let other_fields = fields
+                .named
+                .iter()
+                .filter(|it| libafl_display_type(&it.ty) == TyOther)
+                .map(|it| &it.ident);
+
+            let fmt = " {}";
+            let other_fields_fmt = fmt.repeat(other_fields.clone().count());
+            return TokenStream::from(quote! {
+                impl core::fmt::Display for #ident {
+                    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                        // TyVec
+                        #( for e in &self.#vec_fields {
+                            write!(f, #fmt, e)?;
+                        } )*
+
+                        // TyOption
+                        #( if let Some(opt) = &self.#options_fields {
+                            write!(f, #fmt, opt)?;
+                        } )*
+
+                        // TyOther
+                        write!(f, #other_fields_fmt, #(self.#other_fields), *)?;
+
+                        Ok(())
+                    }
+                }
+            });
+        }
+    }
+    panic!("Only structs are supported");
+}
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Debug, PartialEq, Eq)]
+enum LibaflDisplayFieldType {
```

Comment:
maybe we need something like HashSet as well?
vec is good for a list of devices for example, but does not fit well for a list of unique options.